internal/webhook/general/eviction_webhook.go

```diff
@@ -0,0 +1,286 @@
+	evictionResult := ew.processEvictionRequest(admissionReview, log)
+	if evictionResult != nil {
+		response = *evictionResult
+	}
+
+	// Send final response
+	ew.sendResponse(w, admissionReview, &response)
+	log.Info("Eviction webhook request processed", "allowed", response.Allowed)
+}
+
+// parseAdmissionReview parses the admission review from the request
+func (ew *EvictionWebhook) parseAdmissionReview(r *http.Request) (*admissionv1.AdmissionReview, error) {
+	var admissionReview admissionv1.AdmissionReview
+	if err := json.NewDecoder(r.Body).Decode(&admissionReview); err != nil {
+		return nil, fmt.Errorf("failed to decode admission review: %w", err)
+	}
+
+	if admissionReview.Request == nil {
+		return nil, fmt.Errorf("admission review request is nil")
+	}
+
+	return &admissionReview, nil
+}
+
+// isWebhookEnabled checks if the webhook is enabled via environment variable
+func (ew *EvictionWebhook) isWebhookEnabled() bool {
+	enable, found := os.LookupEnv("ENABLE_SAFE_POD_EVICTION")
+	return found && strings.EqualFold(enable, "true")
+}
+
+// processEvictionRequest processes the eviction request and returns the response
+func (ew *EvictionWebhook) processEvictionRequest(admissionReview *admissionv1.AdmissionReview,
+	log logr.Logger) *admissionv1.AdmissionResponse {
+	// Parse eviction object
+	eviction, err := ew.parseEvictionObject(admissionReview.Request.Object.Raw)
+	if err != nil {
+		log.Error(err, "Failed to parse eviction object")
+
+		return &admissionv1.AdmissionResponse{
+			UID:     admissionReview.Request.UID,
+			Allowed: false,
+			Result: &metav1.Status{
+				Status:  metav1.StatusFailure,
+				Message: fmt.Sprintf("Failed to parse eviction object: %v", err),
+				Code:    http.StatusBadRequest,
+			},
+		}
+	}
+
+	// Get pod information
+	pod, err := ew.getPodForEviction(eviction, admissionReview.Request.Namespace)
+	if err != nil {
+		log.Error(err, "Failed to get pod for eviction", "pod", eviction.Name)
+
+		return &admissionv1.AdmissionResponse{
+			UID:     admissionReview.Request.UID,
+			Allowed: false,
+			Result: &metav1.Status{
+				Status:  metav1.StatusFailure,
+				Message: fmt.Sprintf("Failed to get pod %s/%s: %v", admissionReview.Request.Namespace, eviction.Name, err),
+				Code:    http.StatusInternalServerError,
+			},
+		}
+	}
+
+	// Check if this is an Aerospike pod
+	if !ew.isAerospikePod(pod) {
+		log.V(1).Info("Allowing eviction of non-Aerospike pod", "pod", eviction.Name)
+		return nil // Allow eviction
+	}
+
+	// Block Aerospike pod eviction
+	log.Info("Blocking eviction of Aerospike pod", "pod", eviction.Name)
+
+	// Set annotation asynchronously (non-blocking)
+	// TODO: do we really want async here?
+	go ew.setEvictionBlockedAnnotationAsync(pod)
+
+	return &admissionv1.AdmissionResponse{
+		UID:     admissionReview.Request.UID,
+		Allowed: false,
+		Result: &metav1.Status{
+			Status: metav1.StatusFailure,
+			Message: fmt.Sprintf("Eviction of Aerospike pod %s/%s is blocked by admission webhook",
```

Comment:
Can we make it more descriptive with more details like AKO will take care of pod migration?